helios-ts/src/storage.rs

```diff
@@ -0,0 +1,45 @@
+extern crate console_error_panic_hook;
+extern crate web_sys;
+
+use config::Config;
+use consensus::database::Database;
+use eyre::Result;
+
+#[derive(Clone)]
+pub struct StorageDB {
+    checkpoint: Vec<u8>,
+}
+
+impl Database for StorageDB {
+    fn new(config: &Config) -> Result<Self> {
+        console_error_panic_hook::set_once();
+        let window = web_sys::window().unwrap();
+        if let Ok(Some(local_storage)) = window.local_storage() {
+            let checkpoint = local_storage.get_item("checkpoint").unwrap();
+            let checkpoint = checkpoint
+                .as_ref()
+                .map(|c| c.strip_prefix("0x").unwrap_or(c.as_str()))
+                .map(|c| hex::decode(c).unwrap())
+                .unwrap_or(config.default_checkpoint.clone());
+            return Ok(Self { checkpoint });
+        }
+
+        eyre::bail!("local_storage not available")
+    }
+
+    fn load_checkpoint(&self) -> Result<Vec<u8>> {
+        Ok(self.checkpoint.clone())
```

Comment:
Lets move the fetching logic in the constructor into here so we can ensure we have the freshest checkpoint. It also saves us from needing to store the checkpoint and have an unnecessary clone.

Reply:
sounds good, does this mean we remove the checkpoint field on LocalStorageDB struct, and in constructor, call save_checkpoint to save to actual local_storage, so it can be load later on?

Reply:
I don't think we need to do any saving in the constructor. But yeah we won't need the vec anymore in the struct.